src/apps/basic/basic_apps.lua

```diff
@@ -6,11 +6,40 @@ local app = require("core.app")
 local freelist = require("core.freelist")
 local packet = require("core.packet")
 local link = require("core.link")
+local ffi = require("ffi")
 local transmit, receive = link.transmit, link.receive
 
+--- # `Truncate` app: truncate or zero pad packet to length n
+Truncate = {}
+function Truncate:new(n)
+   return setmetatable({n = n}, {__index=Truncate})
+end
+function Truncate:push()
+   for _ = 1, link.nreadable(self.input.input) do
+      local p = receive(self.input.input)
+      ffi.fill(p.data, math.min(0, self.n - p.length))
```

Comment:
Yeah I know what you mean.  Still, Luke has given several presentations with a lovely slide that's just

```c
// The maximum amount of payload in any given packet.
enum { PACKET_PAYLOAD_SIZE = 10*1024 };

// Packet of network data, with associated metadata.
struct packet {
    unsigned char data[PACKET_PAYLOAD_SIZE];
    uint16_t length;           // data payload length
};
```

and so I dunno, it's part of the public messaging.  There are quite a lot of uses out there already fwiw, and many of them introduced since you made the data/length functions in January last year.